kubeflow/trainer/backends/localprocess/backend.py

```diff
@@ -0,0 +1,236 @@
+        return local_runtimes
+
+    def get_runtime(self, name: str) -> types.Runtime:
+        runtime = next((rt for rt in local_runtimes if rt.name == name), None)
+        if not runtime:
+            raise ValueError(f"Runtime '{name}' not found.")
+
+        return runtime
+
+    def get_runtime_packages(self, runtime: types.Runtime):
+        if isinstance(runtime.trainer, LocalRuntimeTrainer):
+            return runtime.trainer.packages
+        else:
+            logger.debug("Trainer type isn't supported by LocalProcessBackend")
+            return []
+
+    def train(
+        self,
+        runtime: Optional[types.Runtime] = None,
+        initializer: Optional[types.Initializer] = None,
+        trainer: Optional[Union[types.CustomTrainer, types.BuiltinTrainer]] = None,
+    ) -> str:
+        # set train job name
+        train_job_name = random.choice(string.ascii_lowercase) + uuid.uuid4().hex[:11]
+        # localprocess backend only supports CustomTrainer
+        if not isinstance(trainer, types.CustomTrainer):
+            raise ValueError("CustomTrainer must be set with LocalProcessBackend")
+
+        # create temp dir
+        venv_dir = tempfile.mkdtemp(prefix=train_job_name)
+        logger.debug("operating in {}".format(venv_dir))
+
+        runtime.trainer = local_utils.get_runtime_trainer(
+            runtime_name=runtime.name,
+            venv_dir=venv_dir,
+            framework=runtime.trainer.framework,
+        )
+
+        # build training job command
+        training_command = local_utils.get_training_job_command(
+            trainer=trainer,
+            runtime=runtime,
+            train_job_name=train_job_name,
+            venv_dir=venv_dir,
+            cleanup=self.cfg.cleanup_venv,
+        )
+
+        # create subprocess object
+        train_job = LocalJob(
+            name="{}-train".format(train_job_name),
+            command=training_command,
+            execution_dir=venv_dir,
+            env=trainer.env,
+            dependencies=[],
+        )
+
+        self.__register_job(
+            train_job_name=train_job_name,
+            step_name="train",
+            job=train_job,
+            runtime=runtime,
+        )
+        # start the job.
+        train_job.start()
+
+        return train_job_name
+
+    def list_jobs(self, runtime: Optional[types.Runtime] = None) -> List[types.TrainJob]:
+        result = []
+
+        for _job in self.__local_jobs:
+            if runtime and _job.runtime.name != runtime.name:
+                continue
+            result.append(
+                types.TrainJob(
+                    name=_job.name,
+                    creation_timestamp=_job.created,
+                    runtime=runtime,
+                    num_nodes=1,
+                    steps=[
+                        types.Step(name=s.step_name, pod_name=s.step_name, status=s.job.status)
+                        for s in _job.steps
+                    ],
+                )
```

Comment:
Shall we re-use the get_job API here ?
```suggestion
                self.get_job(name=_job.name)
```

Reply:
I think this will create another loop over the `__local_jobs` to get the job.